src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -44,7 +44,9 @@ public ReadPileup(final Locatable loc, final List<PileupElement> pileup) {
      * Note: the current implementation of ReadPileup does not efficiently retrieve the stratified pileup
      */
     public ReadPileup(final Locatable loc, final Map<String, ReadPileup> stratifiedPileup) {
-        this(loc, stratifiedPileup.values().stream().flatMap(ReadPileup::getElementStream).collect(Collectors.toList()));
+        this(loc, stratifiedPileup.values().stream().flatMap(ReadPileup::getElementStream)
+                .sorted((l, r) -> Integer.compare(l.getRead().getStart(), r.getRead().getStart()))
+                .collect(Collectors.toList()));
```

Comment:
Can we do the sort in `FragmentCollection.create()` rather than in the constructor for `ReadPileup`? Does any other code in `ReadPileup` require that the pileup elements be sorted by read start position? If not, I'd prefer not to pay the cost of sorting in the method where it's needed rather than for all `ReadPileups` constructed in this way.